solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -447,24 +447,58 @@ library DepositUtils {
-    function getOwnerRedemptionTbtcRequirement(DepositUtils.Deposit storage _d, address _redeemer) internal view returns(uint256) {
-        // The TDT owner only ever owes fee adjustments.
-        return computeRedemptionFeeAdjustment(_d, _redeemer);
+    function getOwnerRedemptionTbtcRequirement(
+        DepositUtils.Deposit storage _d,
+        address _redeemer
+    ) internal view returns(uint256 owedToDeposit, uint256 owedToRedeemer) {
+        uint256 baseRedemptionRequirement = computeRedemptionFeeAdjustment(_d, _redeemer);
+        uint256 currentBalance = _d.tbtcToken.balanceOf(address(this));
+        if(currentBalance > baseRedemptionRequirement){
+            owedToRedeemer = currentBalance.sub(baseRedemptionRequirement);
+            return(0, owedToRedeemer);
+        }
+        owedToDeposit = baseRedemptionRequirement.sub(currentBalance);
+        return(owedToDeposit, 0);
     }
 
     /// @notice             Get TBTC amount required for redemption by a specified _redeemer.
     /// @dev                Will revert if redemption is not possible by msg.sender.
     /// @param _redeemer    The deposit redeemer.
-    /// @return             The amount in TBTC needed to redeem the deposit.
-    function getRedemptionTbtcRequirement(DepositUtils.Deposit storage _d, address _redeemer) internal view returns(uint256) {
+    /// @return             Tuple, first value represents the amount in TBTC needed to redeem
+    ///                     the deposit. The second value represents the amount in TBTC the deposit
+    ///                     owes to the redeemer in case of overpayment by an external source.
+    function getRedemptionTbtcRequirement(
+        DepositUtils.Deposit storage _d,
+        address _redeemer
+    ) internal view returns(uint256 owedToDeposit, uint256 owedToRedeemer, uint256 owedToTdtHolder) {
         bool redeemerHoldsTdt = depositOwner(_d) == _redeemer;
         bool inCourtesy = _d.inCourtesyCall();
         bool atTerm = remainingTerm(_d) == 0;
 
         require(redeemerHoldsTdt || inCourtesy || atTerm, "Only TDT holder can redeem unless deposit is at-term or in COURTESY_CALL");
 
         uint256 mainCharge = computeBaseRedemptionCharge(_d, redeemerHoldsTdt);
-        return mainCharge.add(computeRedemptionFeeAdjustment(_d, _redeemer));
+        uint256 baseRedemptionRequirement = mainCharge.add(computeRedemptionFeeAdjustment(_d, _redeemer));
+        uint256 currentBalance = _d.tbtcToken.balanceOf(address(this));
+        if(currentBalance == 0){
+            if(!redeemerHoldsTdt){
+                owedToTdtHolder = lotSizeTbtc(_d).sub(signerFee(_d));
+            }
+            return(baseRedemptionRequirement, 0, owedToTdtHolder);
+        }
+        if(!redeemerHoldsTdt){
+            if(currentBalance >= signerFee(_d)){
+                owedToTdtHolder = lotSizeTbtc(_d);
+            }else{
+                owedToTdtHolder = lotSizeTbtc(_d).sub(currentBalance);
```

Comment:
Is this less the current balance, or is it less the difference between the signer fee and the current balance? Intuitively, I think this should be `lotSizeTbtc(_d).sub(signerFee(_d).sub(currentBalance))`; could be wrong.

Reply:
Ah no, I see. Got turned around.

Reply:
I think it should be less the current balance, as long as the current balance is less than the signer fee

Reply:
ah GH didn't refresh to comment, sorry